Core/Core/Inbox/AttachmentPicker/Model/AttachmentPickerInteractor/AttachmentPickerInteractorLive.swift

```diff
@@ -68,10 +106,87 @@ class AttachmentPickerInteractorLive: AttachmentPickerInteractor {
                 uploadManager.cancel(file: file)
             }
         }
+        var filteredFiles: [File] = alreadySelectedFiles.value.filter { $0.isUploading == false}
+        alreadySelectedFiles.send(filteredFiles)
     }
 
     func removeFile(file: File) {
+        if alreadySelectedFiles.value.contains(file) {
+            let newValues = alreadySelectedFiles.value.filter { $0 != file }
+            alreadySelectedFiles.send(newValues)
+        } else {
+            uploadManager.viewContext.delete(file)
+            fileStore.refresh()
+        }
+    }
+
+    func deleteFile(file: File) -> AnyPublisher<Void, Never> {
         uploadManager.viewContext.delete(file)
         fileStore.refresh()
+
+        if let fileId = file.id {
+            return ReactiveStore(useCase: DeleteFile(fileID: fileId))
+                .getEntities()
+                .mapToVoid()
+                .replaceError(with: ())
+                .eraseToAnyPublisher()
+        } else {
+            return Just(()).eraseToAnyPublisher()
+        }
+    }
+
+    private func duplicateFileToUploadFolder(file: File) {
+        if let fileId = file.id {
+            getOnlineFileURL(fileId: fileId)
+                .map { [weak self] url in
+                    if let self, let url {
+                        _ = try? self.uploadManager.add(url: url, batchID: self.batchId)
```

Comment:
Is it okay to do the rest if this one fails? Or it will never fail?

Reply:
I think it's okay in this way. It shouldn't fail but if it does, there is no option to retry it. So by continuing, the user has to select the file again.